inference-engine/src/vpu/graph_transformer/src/middleend/passes/allocate_resources.cpp

```diff
@@ -206,10 +206,32 @@ AllocationResult runAllocator(const Model& model, EnableShapeAllocation enableSh
     //
 
     if (enableShapeAllocation == EnableShapeAllocation::YES) {
-        for (auto data : model->datas()) {
-            const auto shapeLocation = allocator.allocateShape(data);
-            data->setShapeAllocationInfo(shapeLocation);
+        DataSet datasWithAllocatedShape;
```

Comment:
What do you think about adding a private field `DataList _allocatedDataList`? Then:  

```
 _allocatedData.emplace(data);
_allocatedDataList.push_back(data);
```  

As far as I understand, this version has the same amount of memory used, but you don't need to iterate through the stages again.

Reply:
Yes, it's another solution for this problem, but `_allocatedDataList` may confuse the developer on why do we need to insert the same data to different container, of course the appropriate comment may resolve this confusion. For me it's quite equivalent, @gladilov-gleb what do you think?

Reply:
@gladilov-gleb Which one?

Reply:
Doesn't matter

Reply:
The only one concern is the performance. `_allocatedData` is `unordered_set` which is currently used only for inserting and searching the datas. Turning it into a list may give a performance drop even if we don't need to allocate shape at this current allocation run. (Note that we do it only once at the very last allocation). So I still prefer the current solution so it minimally affects the whole algorithm in case we need shape allocation.